packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/documents_batch/action_validation/document_replace_transition_action/state_v0/mod.rs

```diff
@@ -0,0 +1,46 @@
+use dpp::data_contract::accessors::v0::DataContractV0Getters;
+use dpp::identifier::Identifier;
+use dpp::validation::SimpleConsensusValidationResult;
+use drive::state_transition_action::document::documents_batch::document_transition::document_replace_transition_action::{DocumentReplaceTransitionAction, DocumentReplaceTransitionActionAccessorsV0};
+use dpp::version::PlatformVersion;
+use drive::grovedb::TransactionArg;
+use drive::state_transition_action::document::documents_batch::document_transition::document_base_transition_action::DocumentBaseTransitionActionAccessorsV0;
+use crate::error::Error;
+use crate::platform_types::platform::PlatformStateRef;
+
+pub(super) trait DocumentReplaceTransitionActionStateValidationV0 {
+    fn validate_state_v0(
+        &self,
+        platform: &PlatformStateRef,
+        owner_id: Identifier,
+        transaction: TransactionArg,
+        platform_version: &PlatformVersion,
+    ) -> Result<SimpleConsensusValidationResult, Error>;
+}
+impl DocumentReplaceTransitionActionStateValidationV0 for DocumentReplaceTransitionAction {
+    fn validate_state_v0(
+        &self,
+        platform: &PlatformStateRef,
+        owner_id: Identifier,
+        transaction: TransactionArg,
+        platform_version: &PlatformVersion,
+    ) -> Result<SimpleConsensusValidationResult, Error> {
+        let contract_fetch_info = self.base().data_contract_fetch_info();
+
+        let contract = &contract_fetch_info.contract;
+
+        let document_type = contract.document_type_for_name(self.base().document_type_name())?;
+
```

Comment:
We should add a comment that the state validation is actually happening in transform.